qutip/tests/solver/heom/test_bofin_baths.py

```diff
@@ -463,3 +472,247 @@ def test_create(self):
+        assert np.sum(J - J2) < (1e-12)
+        assert self.bath.spec_n <= 5
+
+    def test_matsubara_coefficients_from_spectral_fit(self):
+        ckAR, vkAr, ckAI, vkAI = (
+            [
+                (0.29298628613108785 - 0.2097848947562078j),
+                (0.29298628613108785 + 0.2097848947562078j),
+                (-0.01608448645347298 + 0j),
+                (-0.002015397620259254 + 0j),
+            ],
+            [(1 - 1j), (1 + 1j), (6.283185307179586 + 0j),
+             (12.566370614359172 + 0j)],
+            [(-0 + 0.25j), -0.25j],
+            [(1 - 1j), (1 + 1j)],
+        )
+        self.bath.params_spec = np.array([1]), np.array([1]), np.array([1])
+        self.bath._matsubara_spectral_fit()
+        ckAR2, vkAr2, ckAI2, vkAI2 = self.bath.cvars
+        assert np.isclose(ckAR, ckAR2).all()
+        assert np.isclose(vkAr, vkAr2).all()
+        assert np.isclose(ckAI, ckAI2).all()
+        assert np.isclose(vkAI, vkAI2).all()
+
+
+class TestFitCorr:
+    alpha = 0.005
+    wc = 1
+    T = 1
+    bath = FitCorr(sigmax())
+
+    def test_corr_approx(self):
+        t = np.array([1 / 10, 1 / 5, 1])
+        C = np.array(
+            [
+                -0.900317 + 0.09033301j,
+                -0.80241065 + 0.16265669j,
+                -0.19876611 + 0.30955988j,
+            ]
+        )
+        a, b, c = -np.array([list(range(2))] * 3)
+        C2 = self.bath._corr_approx(t, a, b, c)
+        assert np.isclose(C, C2).all()
+
+    def test_fit_correlation(self):
+        a, b, c = -np.array([list(range(2))] * 3)
+        t = np.linspace(0.1, 10 / self.wc, 1000)
+        C = self.bath._corr_approx(t, a, b, c)
+        self.bath.fit_correlation(t, C, Nr=3, Ni=3)
+        a2, b2, c2 = self.bath.params_real
+        ai2, bi2, ci2 = self.bath.params_imag
+        C2re = self.bath._corr_approx(t, a2, b2, c2)
+        C2imag = self.bath._corr_approx(t, ai2, bi2, ci2)
+        assert np.isclose(np.real(C), np.real(C2re)).all()
+        assert np.isclose(np.imag(C), np.imag(C2imag)).all()
+
+    def test_matsubara_coefficients(self):
+        ans = np.array(
+            [
+                [(0.5 + 0j), (0.5 + 0j), (0.5 - 0j), (0.5 - 0j)],
+                [(-1 - 1j), (-1 - 1j), (-1 + 1j), (-1 + 1j)],
+                [-0.5j, -0.5j, 0.5j, 0.5j],
+                [(-1 - 1j), (-1 - 1j), (-1 + 1j), (-1 + 1j)],
+            ]
+        )
+        self.bath.params_real = [[1] * 2] * 3
+        self.bath.params_imag = [[1] * 2] * 3
+        self.bath._matsubara_coefficients()
+        assert np.isclose(np.array(self.bath.cvars), ans).all()
+
+    def test_power_spectrum(self):
+        self.bath.params_real = [[1] * 2] * 3
+        self.bath.params_imag = [[1] * 2] * 3
+        self.bath._matsubara_coefficients()
+        S = self.bath.power_spectrum(np.array([1]))
+        assert np.isclose(S, np.array([-2.4 + 0j]))
+
+
+@pytest.mark.skipif('mpmath' not in sys.modules,
+                    reason="requires the mpmath library")
```

Comment:
This is a bit weird since `'mpmath'` will not be in `sys.modules` until it is imported, even if it is available. I did some googling, and the recommended way to handle such a situation seems to be `pytest.importorskip`. I don't see a convenient way to apply that to a whole class, the best way seems to be to add `pytest.importorskip("mpmath")` at the beginning of each test in this class.